dbt/config.py

```diff
@@ -23,6 +24,10 @@
 DEFAULT_USE_COLORS = True
 DEFAULT_PROFILES_DIR = os.path.join(os.path.expanduser('~'), '.dbt')
 
+if os.environ.get('DBT_PROFILES_DIR') is not None:
```

Comment:
I think the way this code works, the supplied env var won't have the `~` expanded. That means that quoted vars won't be expanded, whereas unquoted vars will be expanded.

```
$ DBT_PROFILES_DIR="~/.dbt/profiles.yml"
$ echo $DBT_PROFILES_DIR
~/.dbt/profiles.yml

$ DBT_PROFILES_DIR=~/.dbt/profiles.yml
$ echo $DBT_PROFILES_DIR
/Users/drew/.dbt/profiles.yml
```

In the former case, I think dbt will choke on the tilde. Maybe that's just caveat emptor, and a certain amount of bash knowledge is required to use env vars anyway. Alternatively, we could do something like:
```
if os.environ.get('DBT_PROFILES_DIR') is not None:
  PROFILES_DIR = os.path.expanduser(os.environ.get('DBT_PROFILES_DIR'))
```

I don't feel super strongly about this, but wanted to raise the topic.

Reply:
Oh, good point! Combining our two suggestions:
`PROFILES_DIR = os.path.expanduser(os.environ.get('DBT_PROFILES_DIR', DEFAULT_PROFILES_DIR))`

Reply:
^ that's a little funky because we'll be expanding the tilde twice when no env var is present, but I don't think it would be an issue, right?

Reply:
The only way it would be possible to have a problem is if there were a posix system where the home directory started with a literal `~`, in which case the user will have bigger problems.